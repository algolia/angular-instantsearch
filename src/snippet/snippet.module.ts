import { NgModule } from '@angular/core';

import { NgAisSnippet } from './snippet';

@NgModule({
  declarations: [NgAisSnippet],
  exports: [NgAisSnippet],
})
export class NgAisSnippetModule {}
