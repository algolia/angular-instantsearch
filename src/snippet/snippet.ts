import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { Hit } from 'instantsearch.js';
import { snippet } from 'instantsearch.js/es/helpers';

@Component({
  selector: 'ais-snippet',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `<span class="ais-Snippet" [innerHtml]="content"></span>`,
})
export class NgAisSnippet {
  @Input() private readonly attribute: string;
  @Input() private readonly hit: Partial<Hit>;
  @Input() private readonly highlightedTagName: string = 'mark';

  get content() {
    return snippet({
      attribute: this.attribute,
      hit: this.hit,
      highlightedTagName: this.highlightedTagName,
    });
  }
}
