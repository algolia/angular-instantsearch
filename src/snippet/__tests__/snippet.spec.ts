import { Component } from '@angular/core';
import { TestBed } from '@angular/core/testing';

import { NgAisSnippetModule } from '../snippet.module';

const render = ({ hit, attribute, highlightedTagName = 'mark' }) => {
  @Component({
    selector: 'test-component',
    template: `
      <ais-snippet [attribute]="attribute" [hit]="hit" [highlightedTagName]="highlightedTagName">
      </ais-snippet>
    `,
  })
  class TestComponent {
    hit = hit;
    attribute: string = attribute;
    highlightedTagName = highlightedTagName;
  }

  TestBed.configureTestingModule({
    declarations: [TestComponent],
    imports: [NgAisSnippetModule],
  });

  const fixture = TestBed.createComponent(TestComponent);
  fixture.detectChanges();

  return fixture;
};

describe('highlight', () => {
  it('should snippet strings', () => {
    const fixture = render({
      attribute: 'description',
      hit: {
        _snippetResult: {
          description: { value: '… <mark>foo</mark> bar …' },
        },
      },
    });
    expect(fixture).toMatchSnapshot();
  });

  it('should snippet with a custom highlightedTagName', () => {
    const fixture = render({
      attribute: 'description',
      hit: {
        _snippetResult: {
          description: { value: '… <mark>foo</mark> bar …' },
        },
      },
      highlightedTagName: 'em',
    });
    expect(fixture).toMatchSnapshot();
  });
});
