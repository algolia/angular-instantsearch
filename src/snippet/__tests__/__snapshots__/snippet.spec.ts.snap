// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight should snippet strings 1`] = `
<test-component
  attribute={[Function String]}
  highlightedTagName={[Function String]}
  hit={[Function Object]}
>
  <ais-snippet>
    <span
      class="ais-Snippet"
    >
      … 
      <mark
        class="ais-Snippet-highlighted"
      >
        foo
      </mark>
       bar …
    </span>
  </ais-snippet>
</test-component>
`;

exports[`highlight should snippet with a custom highlightedTagName 1`] = `
<test-component
  attribute={[Function String]}
  highlightedTagName={[Function String]}
  hit={[Function Object]}
>
  <ais-snippet>
    <span
      class="ais-Snippet"
    >
      … 
      <em
        class="ais-Snippet-highlighted"
      >
        foo
      </em>
       bar …
    </span>
  </ais-snippet>
</test-component>
`;
